addons/binding/org.openhab.binding.rme/src/main/java/org/openhab/binding/rme/handler/SerialThingHandler.java

```diff
@@ -0,0 +1,238 @@
+ * Configuration of the underlying Thing, if not already specified in the
+ * thing.xml definition
+ * 
+ * @author Karel Goderis - Initial contribution
+ */
+public abstract class SerialThingHandler extends BaseThingHandler implements
+		SerialPortEventListener {
+
+	// List of all Configuration parameters
+	public static final String PORT = "port";
+	public static final String BAUD_RATE = "baud";
+	public static final String BUFFER_SIZE = "buffer";
+
+	private Logger logger = LoggerFactory.getLogger(SerialThingHandler.class);
+
+	private SerialPort serialPort;
+	private CommPortIdentifier portId;
+	private InputStream inputStream;
+	private OutputStream outputStream;
+
+	public SerialThingHandler(Thing thing) {
+		super(thing);
+	}
+
+	/**
+	 * Called when data is received on the serial port
+	 * 
+	 * @param line
+	 *            - the received data as a String
+	 * 
+	 **/
+	abstract public void onDataReceived(String line);
+
+	/**
+	 * Write data to the serial port
+	 * 
+	 * @param msg
+	 *            - the received data as a String
+	 * 
+	 **/
+	public void writeString(String msg) {
+		String port = (String) this.getConfig().get(PORT);
+
+		try {
+			// write string to serial port
+			outputStream.write(msg.getBytes());
+			outputStream.flush();
+		} catch (IOException e) {
+			logger.error("Error writing '{}' to serial port {}: {}",
+					new Object[] { msg, port, e.getMessage() });
+		}
+	}
+
+	@Override
+	public void serialEvent(SerialPortEvent event) {
+
+		switch (event.getEventType()) {
+		case SerialPortEvent.BI:
+		case SerialPortEvent.OE:
+		case SerialPortEvent.FE:
+		case SerialPortEvent.PE:
+		case SerialPortEvent.CD:
+		case SerialPortEvent.CTS:
+		case SerialPortEvent.DSR:
+		case SerialPortEvent.RI:
+		case SerialPortEvent.OUTPUT_BUFFER_EMPTY:
+			break;
+		case SerialPortEvent.DATA_AVAILABLE:
+			try {
+				BufferedReader br = new BufferedReader(new InputStreamReader(
+						inputStream), ((int) getConfig().get(BUFFER_SIZE)));
+				while (br.ready()) {
+					String line = br.readLine();
+					logger.debug("Receving '{}' on '{}'", line,
+							(String) getConfig().get(PORT));
+					onDataReceived(line);
+				}
+			} catch (IOException e) {
+				String port = (String) getConfig().get(PORT);
+				logger.debug("Error receiving data on serial port {}: {}",
+						new Object[] { port, e.getMessage() });
+			}
+			break;
+		}
+	}
+
+	@Override
+	public void dispose() {
+		logger.debug("Disposing serial thing handler.");
+		IOUtils.closeQuietly(inputStream);
+		IOUtils.closeQuietly(outputStream);
+		serialPort.close();
+	}
+
+	@Override
+	public void initialize() {
+		logger.debug("Initializing serial thing handler.");
+
+		String port = (String) getConfig().get(PORT);
+		int baud = (int) getConfig().get(BAUD_RATE);
+
+		if (serialPort == null) {
+
+			// parse ports and if the default port is found, initialized the
+			// reader
+			@SuppressWarnings("rawtypes")
+			Enumeration portList = CommPortIdentifier.getPortIdentifiers();
```

Comment:
We should think about some convenience methods in io.transport.serial for such things.